### PR TITLE
Updating pipeline to run E2E tests for ooproc

### PR DESCRIPTION
**Pull request by aishwaryabh**
Resolves: https://github.com/Azure/azure-functions-pyfx-planning/issues/151

This PR addresses running the E2E tests for the chat completion dotnet isolated scenario. In order for isolated to work, the worker needs to build its own local package of web jobs to pick up the latest changes. The pipeline creates a test package called `Microsoft.Azure.WebJobs.Extensions.OpenAI.99.99.99-test.nupkg` and puts it into a local feed called `NuGetPackagesLocal`. The worker then consumes these changes and runs the E2E tests as expected.

I also removed the in-proc sample project and tests for chat completion, since going forward we would want users to only use the isolated model.

### CODE REVIEW

**Comment by manvkaur** [comment]
try providing model name like it is done in in proc sample to run these tests - https://github.com/Azure/azure-functions-openai-extension/blob/main/samples/chat/csharp-inproc/ChatBot.cs#L43C8-L43C113
 
the tests run on Azure Open AI, it expects the deployment name as model name.

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -10,6 +10,7 @@ on:
 
 env:
   config: Release
+  newVersion: 99.99.99-test
```
Comment (cgillum) [inline .github/workflows/build.yml:13]: Maybe name this `fakeWebJobsPackageVersion` or something that makes it more clear that this isn't a real version number?

File: .github/workflows/build.yml [hunk 2]
```diff
@@ -23,45 +24,52 @@ jobs:
...
-      run: docker build -f samples/chat/csharp-inproc/Dockerfile -t openai-func-sample-csharp-inproc .
+    - name: Docker Build - Chat Bot Sample Out of proc
+      run: docker build -f samples/chat/csharp-ooproc/Dockerfile -t openai-func-sample-csharp-ooproc .
 
     - name: Docker Run - Azurite
       run: docker run -d -p 10000:10000 -p 10001:10001 -p 10002:10002 mcr.microsoft.com/azure-storage/azurite
 
-    - name: Docker Run - Chat Bot Sample
+    - name: Docker Build - Chat Bot Sample Out of proc
+      run: docker build -f samples/chat/csharp-ooproc/Dockerfile -t openai-func-sample-csharp-ooproc .
```
Comment (cgillum) [inline .github/workflows/build.yml:34]: Not sure if I'm reading this diff correctly, but it looks like we have two duplicate `docker build` steps?

File: samples/chat/csharp-ooproc/ChatBotIsolated.cs [hunk 3]
```diff
@@ -79,7 +78,7 @@ public class CreateChatBotOutput
 
     public class PostResponseOutput
     {
-        [ChatBotPostOutput("{chatId}", Model = "gpt-3.5-turbo")]
+        [ChatBotPostOutput("{chatId}", Model = "%AZURE_DEPLOYMENT_NAME%")]
```
Comment (cgillum) [inline samples/chat/csharp-ooproc/ChatBotIsolated.cs:81]: Is this change necessary? If so, do we need to make any corresponding changes to the README to account for this new environment variable requirement when people run the sample locally?

File: samples/chat/csharp-ooproc/Dockerfile [hunk 4]
```diff
@@ -0,0 +1,26 @@
+FROM mcr.microsoft.com/dotnet/sdk:7.0 AS build-env
```
Comment (cgillum) [inline samples/chat/csharp-ooproc/Dockerfile:1]: I recommend we use 8.0 instead of 7.0.

File: src/Functions.Worker.Extensions.OpenAI/Functions.Worker.Extensions.OpenAI.csproj [hunk 5]
```diff
@@ -3,11 +3,18 @@
...
   <ItemGroup>
     <PackageReference Include="Azure.AI.OpenAI" Version="1.0.0-beta.13" />
     <PackageReference Include="Microsoft.Azure.Functions.Worker.Extensions.Abstractions" Version="1.3.0" />
   </ItemGroup>
+
+  <ItemGroup>
+    <AssemblyAttribute Include="Microsoft.Azure.Functions.Worker.Extensions.Abstractions.ExtensionInformationAttribute">
+      <_Parameter1>Microsoft.Azure.WebJobs.Extensions.OpenAI</_Parameter1>
+      <_Parameter2>$(WebJobsVersion)</_Parameter2>
+    </AssemblyAttribute>
```
Comment (cgillum) [inline src/Functions.Worker.Extensions.OpenAI/Functions.Worker.Extensions.OpenAI.csproj:18]: Cool! I didn't know we could do something like this! I have some other projects where this will be useful...

File: .github/workflows/build.yml [hunk 6]
```diff
@@ -23,45 +24,52 @@ jobs:
...
+
+    - name: Create/Copy NuGet Package to local directory and Build
+      run: |
+        dotnet build /home/runner/work/azure-functions-openai-extension/azure-functions-openai-extension/src/WebJobs.Extensions.OpenAI/WebJobs.Extensions.OpenAI.csproj --configuration $config -p:FileVersionRevision=$GITHUB_RUN_NUMBER -p:WebJobsVersion=$newVersion -p:Version=$newVersion
+        sudo cp "/home/runner/work/azure-functions-openai-extension/azure-functions-openai-extension/src/WebJobs.Extensions.OpenAI/bin/Release/Microsoft.Azure.WebJobs.Extensions.OpenAI.${newVersion}.nupkg" ~/NuGetPackagesLocal
+
+        dotnet nuget add source ~/NuGetPackagesLocal
+        dotnet nuget list source
+
+        dotnet build --configuration $config /p:WebJobsVersion=$newVersion;FileVersionRevision=$GITHUB_RUN_NUMBER
```
Comment (cgillum) [inline .github/workflows/build.yml:64]: Given that the Docker build has already succeeded in the previous step, is there a reason for doing a separate build here in the CI process?

**Review by cgillum — COMMENTED** [review]
Awesome to see this! A few minor questions/comments below.

File: OpenAI-Extension.sln [hunk 7]
```diff
@@ -22,8 +22,6 @@ Project("{2150E333-8FDC-42A3-9474-1A3956D46DE8}") = "samples", "samples", "{865C
 EndProject
 Project("{2150E333-8FDC-42A3-9474-1A3956D46DE8}") = "chat", "chat", "{46888A9D-8058-499C-9F2F-07800FF686EB}"
 EndProject
-Project("{9A19103F-16F7-4668-BE54-9A1E7A4F7556}") = "ChatBotSample", "samples\chat\csharp-inproc\ChatBotSample.csproj", "{2C15A452-14A2-4156-95E2-E6D709F383EF}"
```
Comment (manvkaur) [inline OpenAI-Extension.sln:25]: the samples are still existing in the repo, just removed from the solution, need to delete that folder manually and any references in the Readme should be updated too

File: samples/chat/csharp-ooproc/ChatBotIsolated.cs (hunk 3 above)
Comment (aishwaryabh) [reply samples/chat/csharp-ooproc/ChatBotIsolated.cs:81]: Yeah we do need this variable for running the E2E test. This is similar to how the in-proc Chat completion E2E tests were running, in which we need to specify the azure deployment name. I'll update the read me with this info!

File: .github/workflows/build.yml (hunk 1 above)
Comment (aishwaryabh) [reply .github/workflows/build.yml]: done!

File: .github/workflows/build.yml (hunk 2 above)
Comment (aishwaryabh) [reply .github/workflows/build.yml]: Thanks for catching that! Removed second docker build step

File: samples/chat/csharp-ooproc/ChatBotIsolated.cs (hunk 3 above)
Comment (aishwaryabh) [reply samples/chat/csharp-ooproc/ChatBotIsolated.cs:81]: Yeah I do think we need this change since the E2E tests rely on the azure deployment name within the model, similar to how the E2E tests were working the csharp in-proc model. I went ahead and updated the README

File: .github/workflows/build.yml (hunk 6 above)
Comment (aishwaryabh) [reply .github/workflows/build.yml]: Yeah so I tried it without the separate build in the CI process, but it fails to pick up any changes within webjobs. For some reason we need both the docker changes and the CI changes. We probably need both since the docker build changes are within the container only and not within the actual build pipeline, so we need to explicitly build and add the nuget source again.

File: OpenAI-Extension.sln (hunk 7 above)
Comment (aishwaryabh) [reply OpenAI-Extension.sln:25]: Removed it and updated read me!

**Review by manvkaur — APPROVED** [review]
LGTM

File: .github/workflows/build.yml (hunk 6 above)
Comment (cgillum) [reply .github/workflows/build.yml]: I guess it's because the test project, which is built in the CI process, depends on the built extension.

File: samples/chat/README.md [hunk 8]
```diff
@@ -37,10 +36,10 @@ Please refer to the root level [README](../../README.md#requirements) for prereq
         GetChatState: [GET] http://localhost:7071/api/chats/{chatId}
 
         PostUserResponse: [POST] http://localhost:7071/api/chats/{chatId}
-
-        OpenAI::ChatBotEntity: entityTrigger
     ```
 
+    Note for running the post user response function provided in the sample, please specify a model name in the environment variable `AZURE_DEPLOYMENT_NAME`. This value can be an Azure deployment name or a GPT model name.
```
Comment (cgillum) [inline samples/chat/README.md:41]: How do users know what value to provide for `AZURE_DEPLOYMENT_NAME`? Can we provide an example? Also, if this is required for the sample, then we should add it with a default value in the local.settings.json file.

**Review by cgillum — CHANGES_REQUESTED** [review]
One small blocker for me to make it easier for users to successfully run the sample.

File: samples/chat/README.md (hunk 8 above)
Comment (aishwaryabh) [reply samples/chat/README.md]: Updated the documentation! Let me know if this makes sense

**Review by cgillum — APPROVED** [review]
